packages/fs-detectors/src/detectors/filesystem.ts

```diff
@@ -96,12 +96,13 @@ export abstract class DetectorFilesystem {
   /**
    * Writes a file to the filesystem cache.
    * @param name the name of the file to write
-   * @param content The content of the file
+   * @param exists true if the files exists, false otherwise
+   * @param content contents of the file
    */
-  public writeFile(name: string, content?: string): void {
-    if (content)
+  public writeFile(name: string, exists?: boolean, content?: string): void {
+    if (content !== undefined && exists)
       this.readFileCache.set(name, Promise.resolve(Buffer.from(content)));
-    this.fileCache.set(name, Promise.resolve(true));
-    this.pathCache.set(name, Promise.resolve(true));
+    if (exists !== undefined) this.fileCache.set(name, Promise.resolve(exists));
```

Comment:
could these go in one if block?

Reply:
Went in another direction with the writeFile signature - these were removed 🥳